src/sentry/static/sentry/app/views/eventsV2/eventView.tsx

```diff
@@ -743,12 +761,16 @@ class EventView {
     return newEventView;
   }
 
-  withDeletedColumn(columnIndex: number, tableDataMeta: MetaType): EventView {
+  withDeletedColumn(columnIndex: number, tableDataMeta: MetaType | undefined): EventView {
     // Disallow removal of the orphan column, and check for out-of-bounds
     if (this.fields.length <= 1 || this.fields.length <= columnIndex || columnIndex < 0) {
       return this;
     }
 
+    // ensure tableDataMeta is non-empty
+    tableDataMeta =
+      tableDataMeta && Object.keys(tableDataMeta).length > 0 ? tableDataMeta : undefined;
```

Comment:
Should we have a helper function in this module for this check as it occurs a few times?

Reply:
Done in https://github.com/getsentry/sentry/pull/15774/commits/0ceec06cdfa0b2b34f22c2b7655abffcbafd35ca